packages/playground/src/components/Playground.tsx

```diff
@@ -0,0 +1,357 @@
+  const [validator, setValidator] = useState<string>('AJV8');
+  const [showForm, setShowForm] = useState(false);
+  const [liveSettings, setLiveSettings] = useState({
+    showErrorList: 'top',
+    validate: false,
+    disable: false,
+    readonly: false,
+    omitExtraData: false,
+    liveOmit: false,
+  });
+  const [FormComponent, setFormComponent] = useState<any>(withTheme({}));
+  const [ArrayFieldTemplate, setArrayFieldTemplate] = useState<any>();
+  const [ObjectFieldTemplate, setObjectFieldTemplate] = useState<any>();
+
+  const playGroundFormRef = useRef<any>(null);
+
+  useEffect(() => {
+    const hash = document.location.hash.match(/#(.*)/);
+    if (hash && typeof hash[1] === 'string' && hash[1].length > 0) {
+      try {
+        load(JSON.parse(atob(hash[1])));
+      } catch (err) {
+        alert('Unable to load form setup data.');
+      }
+    } else {
+      // initialize theme
+      onThemeSelected(theme, themes[theme]);
+
+      setShowForm(true);
+    }
+  }, []);
+
+  const load = (data: any) => {
+    // Reset the ArrayFieldTemplate whenever you load new data
+    const { ArrayFieldTemplate, ObjectFieldTemplate, extraErrors } = data;
+    // uiSchema is missing on some examples. Provide a default to
+    // clear the field in all cases.
+    const { uiSchema = {} } = data;
+
+    const { theme: dataTheme = theme } = data;
+
+    onThemeSelected(dataTheme, themes[dataTheme]);
+
+    // force resetting form component instance
+    setShowForm(false);
+
+    setUiSchema(uiSchema);
+    setExtraErrors(extraErrors);
+    setTheme(dataTheme);
+    setArrayFieldTemplate(ArrayFieldTemplate);
+    setObjectFieldTemplate(ObjectFieldTemplate);
+    setShowForm(true);
+  };
+
+  const onThemeSelected = (theme: string, { stylesheet, theme: themeObj }: { stylesheet?: any; theme?: any }) => {
+    setTheme(theme);
+    setSubtheme(null);
+    setFormComponent(withTheme(themeObj));
+    setStylesheet(stylesheet);
+  };
+
+  const onSubthemeSelected = (subtheme: any, { stylesheet }: { stylesheet: any }) => {
+    setSubtheme(subtheme);
+    setStylesheet(stylesheet);
+  };
+
+  const onValidatorSelected = (validator: string) => {
+    setValidator(validator);
+  };
+
+  const handleSetLiveSettings = ({ formData }: { formData: any }) => {
+    setLiveSettings(formData);
+  };
+
+  const onFormDataChange = ({ formData = '' }, id: string) => {
+    if (id) {
+      console.log('Field changed, id: ', id);
+    }
+
+    setFormData(formData as any);
+    setShareURL(null);
+  };
+
+  const onShare = () => {
+    const {
+      location: { origin, pathname },
+    } = document;
+    try {
+      const hash = btoa(
+        JSON.stringify({
+          formData,
+          schema,
+          uiSchema,
+          theme,
+          liveSettings,
+        })
+      );
+
+      setShareURL(`${origin}${pathname}#${hash}`);
+    } catch (err) {
+      setShareURL(null);
+    }
+  };
+
+  let templateProps: any = {};
```

Comment:
Does this work?
```suggestion
  let templateProps: Partial<TemplatesType> = {};
```

Reply:
Gives `Property 'extraErrors' does not exist on type 'Partial<TemplatesType<any, RJSFSchema, any>>'.`. Maybe we should update TemplatesType itself?

Reply:
`extraErrors` is not a template thing, it's a Form prop. I wonder why we are storing it there? A work-around is to do the following or just put `extraErrors` in the right spot.
```
let templateProps: Partial<TemplatesType> && { extraErrors?: ErrorSchema };
```

Reply:
Yeah, just remove `extraErrors` and include it explicitly on the `Form` below

Reply:
Now FormComponent is mad:

```
No overload matches this call.
  Overload 2 of 2, '(props: PropsWithChildren<FormProps<any, RJSFSchema, any>>, context?: any): ReactElement<any, any> | Component<FormProps<any, RJSFSchema, any>, any, any> | null', gave the following error.
    Type '({ formData }: { formData?: string | undefined; }, id: string) => void' is not assignable to type '(data: IChangeEvent<any, RJSFSchema, any>, id?: string | undefined) => void'.
  Overload 2 of 2, '(props: PropsWithChildren<FormProps<any, RJSFSchema, any>>, context?: any): ReactElement<any, any> | Component<FormProps<any, RJSFSchema, any>, any, any> | null', gave the following error.
    Type '({ formData }: { formData: string; }, e: any) => void' is not assignable to type '(data: IChangeEvent<any, RJSFSchema, any>, event: FormEvent<any>) => void'.ts(2769)
index.d.ts(52, 5): The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & FormProps<any, RJSFSchema, any> & { children?: ReactNode; }'
index.d.ts(61, 5): The expected type comes from property 'onSubmit' which is declared here on type 'IntrinsicAttributes & FormProps<any, RJSFSchema, any> & { children?: ReactNode; }'
```

Reply:
What happens when you fix the types of the `onSubmit` and `onChange` to match what is expected?

Reply:
Fixed it. Thanks